frontend/actions.js

```diff
@@ -26,32 +31,71 @@ export default ({setState, getState}) => {
       optionalArgsObj
     )
 
-  const loadWalletFromMnemonic = async (state, mnemonic) => {
-    setState(loadingAction(state, 'Loading wallet data...'))
+  const loadWalletFromMnemonic = async (state) => {
+    try {
+      setState(Object.assign(loadingAction(state, 'Loading wallet data...'), {mn: undefined}))
+
+      wallet = await Cardano.CardanoWallet({
+        cryptoProvider: 'mnemonic',
+        mnemonicOrHdNodeString: state.mnemonic,
+        config: CARDANOLITE_CONFIG,
+      })
+
+      const walletIsLoaded = true
+      const ownAddresses = await wallet.getOwnAddresses()
+      const transactionHistory = await wallet.getHistory()
+      const balance = await wallet.getBalance()
+      const sendAmount = {fieldValue: ''}
+      const sendAddress = {fieldValue: ''}
+      const sendResponse = ''
+      setState({
+        walletIsLoaded,
+        ownAddresses,
+        balance,
+        sendAmount,
+        sendAddress,
+        sendResponse,
+        transactionHistory,
+        loading: false,
+        mnemonic: '',
+      })
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e.toString())
+
+      setState({
+        mnemonicValidationError: {
+          code: 'WalletInitializationError',
+        },
+      })
+    }
+  }
+
+  const loadDemoWallet = (state) => {
+    setState({
+      mnemonic: 'civil void tool perfect avocado sweet immense fluid arrow aerobic boil flash',
```

Comment:
I would change this as you suggested - get it from `CARDANOLITE_CONFIG` so developers can change it during development